### ISSUE TITLE
Voicemail default recording says extension number, I don't want that

### ISSUE THREAD

**Issue by nexusxo** [fusionpbx/fusionpbx#2715]
Hello,
I have researched this for multiple days and only found one post and it doesn't seem to work. The default voicemail says, "the caller you reached at extension 1000 (ext #) is not available" and I want it to say, "the caller you reached at extension 1234567890 (the number dialed) is not available". So, I already went into the sounds.xml in the templates/en/vm folder and changed data=$1 to data=${destination_number} and I reloaded the xml files and I restarted the server. It is always the same 1000. I am calling from an outside phone. Why is my change to sounds.xml not taking effect? I also have the lastest FusionPBX/FS released. Any help, advice, guidance would be greatly appreciated!

Thanks!

**Comment by ruslanguns** [comment]
Did you tried stoping FS services and try again?


2017-06-26 21:41 GMT-06:00 nexusxo <notifications@github.com>:

**Comment by nexusxo** [comment]
Thanks for your reply! I have rebooted the whole server after the change. Would that achieve the stopping of FS service? If I didn't understand you question, could you explain just a bit more. Thank you again!!

**Comment by DigiDaz** [comment]
${destination_number} IS the extension number, remember, this is a transfer so if you transfer to 100 the destination_number IS 100.

At some stage earlier in the call, probably at inbound route stage you need to take the destination number there and put it in another variable eg dialled_number. Then you would need to export this variable to the other leg.

**Comment by DigiDaz** [comment]
To cure all of this, why do you not just get the user to set up their mailbox properly so it instead says "Joe Bloggs is not available"

**Comment by nexusxo** [comment]
DigiDaz, thank you for the reminder about 100 being the destination. I was thinking that it stored the original number. Ok, so now where to begin on the variable, this will be fun! I would love for people to setup their voicemail box, but I have no way to make them and I still know people that have not setup their cellphone voicemail for over 2 years, lol! This is for residential service customers.

**Comment by DigiDaz** [comment]
well you can start in the inbound route with something similar to:

action export dialled_number=${destination_number}

**Comment by nexusxo** [comment]
Thank you for your help! I know php really well and I install 3CX phone systems, so I know voip and programming, but fairly new to FS. I greatly appreciate everyone's comments and help!

**Comment by moteus** [comment]
Voice mail stuff wirtten on Lua.
I think interested part is in [macro.lua](https://github.com/fusionpbx/fusionpbx/blob/master/resources/install/scripts/app/voicemail/resources/functions/macro.lua#L44-L50)
You can e.g. add new option via session variable and add check there.

**Comment by nexusxo** [comment]
Thank you @moteus, but the part you pointed out in the macro.lua, I changed the data= section to "1234" instead of a variable just to test it. Reloadxml and restarted FS, but it still said ext 1000. I'm not sure why no matter what file I change, there seems to be no effect. I greatly appreciate everyone's help!! 

@DigiDaz - When you said "well you can start in the inbound route with something similar to:" are you referring to an xml or php file? Could you reference a file location, if possible.

**Comment by DigiDaz** [comment]
In the gui, inbound route.

Also, never mind reloading xml files, you shouldn't be touching them, the xml comes from the db. What you do need to do is flush memcache after changing things.

**Comment by moteus** [comment]
First. FS/Fusion do not cache Lua so you do not need restart anything after change Lua code.
Second. You need edit file in FS script directory. Fusion copy all lua files to there when do Upgrage/ app default. (and in some other cases)
So edit file there is temporary and for test purposes only.

**Comment by nexusxo** [comment]
Ok, great!! Thank you!! I am finally getting somewhere. @DigiDaz : I created the inbound rule like you said and it is as follows: "action export called_number=${destination_number} 0 25". @moteus I edited the script like you said. So in macro.lua I changed to : "table.insert(actions, app="say.number.iterated",data=**called_number**});" and used the called_number variable I created in the inbound route. So now when I call, it is just not saying an ext, like the variable is equal to nothing. Why would the newly created called_number not have a value? Am I calling it incorrectly in lua? I can't say thank you enough for your help! I was about to pull my hair out.

**Comment by DigiDaz** [comment]
You should see the variable in the fs_cli log

**Comment by nexusxo** [comment]
Yes, @DigiDaz I do. I shows [called_number]=[+1352XXXXXXX]. So the question is, why will lua not read it. Would the "+" sign mess lua up, or would that just be skipped and start with the 1?

**Comment by DigiDaz** [comment]
are you actually doing a session get on that variable?? You need to pull it out

**Comment by nexusxo** [comment]
In macro.lua I have "table.insert(actions, app="say.number.iterated",data=**called_number**})" as all other variables in the file do not have a ${}. So I was just calling it like called_number, but I did try : "table.insert(actions, app="say.number.iterated",data=**"${called_number}"**})" with no success.

**Comment by nexusxo** [comment]
I added "called_number = session:getVariable("called_number");" to the if statement, but still have no value for the variable.
